src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -474,6 +479,9 @@ def __deepcopy__(self, memodict={}) -> "SnowflakePlan":  # noqa: B006
             # session object after deepcopy
             session=self.session,
         )
+        copied_plan._is_valid_for_replacement = True
+        if copied_source_plan:
```

Comment:
i am little bit worried that if the source plan is not in the format that one node is logical plan node, another node is snwoflake plan node, some node may not be marked correctly. However, let's keep it like this for now. we can remove this later if we run into problem